enhancements/profile-versioning.md

```diff
@@ -0,0 +1,245 @@
+data stream contains `metadata.version`, the profile parser will persist the
+`Profile` CRD with a version attribute set.
+
+Because the profile version is ultimately coming from the content, we need to
+have a process or automation in place to ensure the profile version is accurate
+across releases.
+
+### User Stories
+
+1. As a cluster administrator, I want to be able to see the version of a
+   particular profile supported by the Compliance Operator.
+2. As a cluster administrator, I want to be able to see if the content used
+   across Compliance Operator releases changed based on the `version`.
+
+### API Extensions
+
+This enhancement focuses solely on adding a `version` property to the `Profile`
+CRD. No other resources will be modified as a result of this implementation.
+
+The implementation should be backwards compatible so that if a profile does not
+have a version the `Profile` will have an empty value for `version`.
+
+### Version Evaluation
+
+Profile version semantics should follow [semantic versioning v2.0.0](https://semver.org/spec/v2.0.0.html),
+which is the most current supported version of SemVer at the time of this
+writing. A summary of the differences between version 1.0.0 and 2.0.0 are
+available in a [GitHub issue](https://github.com/semver/semver/issues/231).
+
+This would allow us to use the following strings to represent profile versions:
+
+* `1.2.0` being the most basic form of versions we see in compliance benchmarks
+* `1.2.0-1` includes flexibility to denote differences across the same profile version
+* `1.2.0-1+e076b1d7ac` include the content of the profile as a SHA
+
+Initially, we may choose to only implement simple versions (e.g., 1.2.0, or
+4.0.0). Depending on user feedback, we have the option to include support for
+more complex version strings that use pre-release or build metadata within the
+semantic version specification to denote changes between operator releases.
+
+Note that if we do decide to support pre-release or build metadata as part of
+the version string, we should ensure version string evaluation considers
+`1.2.0` older than `1.2.0-1` or `1.2.0-2+e076b1d7ac`.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Should the `version` attribute be immutable?
+2. Should a profile have multiple versions (e.g., PCI-DSS version 3.2.1 and 4.0.0)?
```

Comment:
The easy answer I think would be to maintain multiple profiles, each with a different version. Otherwise, we would need to see if the data stream would support a profile with multiple versions, and then ensure the Profile CRDs advertise the various versions available during use. For example, it might be the difference between:

```console
$ oc get profiles.compliance 
NAME       VERSION
pci-dss    3.2.1
pci-dss    4.0.0
ocp4-cis   1.4.0
```


```console
$ oc get profiles.compliance 
NAME 
pci-dss 
ocp4-cis
$ oc get profiles.compliance pci-dss 
NAME       VERSIONS
pci-dss    3.2.1, 4.0.0
```

I'm not sure if the ComplianceAsCode/content build tooling would support the second case, but @yuumasato might know.

Reply:
I think we will have issues here because two CRD can't share same name?

Reply:
I think the idea of
```
oc get profiles.compliance pci-dss 
NAME       VERSIONS
pci-dss    3.2.1, 4.0.0
```
is good here, maybe we can have multiple version xml config map files in the future, and provide a default version for user, and have an option to choose a specific version